app/src/main/java/org/wikipedia/main/MainActivity.kt

```diff
@@ -83,12 +81,16 @@ class MainActivity : SingleFragmentActivity<MainFragment>(), MainFragment.Callba
         override fun onReceive(context: Context?, intent: Intent?) {
             if (intent != null && context != null) {
                 if (intent.action == ENVOY_BROADCAST_VALIDATION_SUCCEEDED) {
-                    val validUrl = intent.getStringExtra(ENVOY_DATA_URL_SUCCEEDED)
-                    val validService = intent.getStringExtra(ENVOY_DATA_SERVICE_SUCCEEDED)
+                    val validUrl = intent.getStringExtra(ENVOY_DATA_URL_SUCCEEDED) ?: ""
+                    var validService = intent.getStringExtra(ENVOY_DATA_SERVICE_SUCCEEDED) ?: ""
+                    if (validUrl.startsWith("envoy") && validService.startsWith("http")) {
```

Comment:
why is the service http if the url is envoy?

Reply:
just to include this here: they're handled the same way in envoy so they get lumped together, but now i need to make the distinction for the revised analytics